src/models/continuity/ContGateTurn.ts

```diff
@@ -30,17 +32,56 @@ export default class ContGateTurn
 
   getHumanReadableText(): string {
     if (this.humanReadableText !== "") return this.humanReadableText;
-    // TODO: Implement
-    return "";
+    const rotation: string = Math.sign(this.angle) === 1 ? "CW" : "CCW";
+    return `GT${this.marchType} ${this.duration} COUNTS ${Math.abs(
+      this.angle
+    )} DEGREES ${rotation} ABOUT POINT [${this.centerPoint}]`;
   }
 
-  /* eslint-disable @typescript-eslint/no-unused-vars */
-  addToFlow(
-    flow: FlowBeat[],
-    startDot: StuntSheetDot,
-    endDot?: StuntSheetDot
-  ): void {
-    // TODO: Implement
+  addToFlow(flow: FlowBeat[], startDot: StuntSheetDot): void {
+    const [startx, starty]: [number, number] = startPositionHelper(
+      flow,
+      startDot
+    );
+    const [dx, dy]: [number, number] = [
+      startx - this.centerPoint[0],
+      starty - this.centerPoint[1],
+    ];
+    if (dx !== 0 || dy !== 0) {
+      for (let beat = 1; beat <= this.duration; beat += 1) {
+        const theta: number =
+          (((beat - 1) / this.duration) * this.angle * Math.PI) / 180;
+        const x: number = dx * Math.cos(theta) + dy * Math.sin(theta);
+        const y: number = dx * Math.sin(theta) + dy * Math.cos(theta);
+        const direction: number =
+          ((Math.sign(this.angle) > 0 ? 180 : 360) -
+            (Math.atan2(-y, x) * 180) / Math.PI) %
+          360;
```

Comment:
Can you leave a comment to explain how the math works here? I'm very rusty on my geometry haha :laughing: 

A wikipedia link works as well!

Reply:
Yes! 
First part is the theta calculation, which is just found by dividing the total rotation linearly based on the total duration and converting to radians.
Next is the dx, dy stuff, which is the rotation about the center point. This is similar to https://en.wikipedia.org/wiki/Rotation_matrix, but with a negative theta and y. I went ahead and proved it here to make sure:
<img width="498" alt="image" src="https://user-images.githubusercontent.com/29964426/103452802-15cdbc80-4c88-11eb-8838-8a1bcad08a2b.png">

The direction is the tangent of that point from the unit circle. Normally that would be just the theta + pi/2, but for some reason I was getting the wrong answer from that so I used the arctangent adding pi/2 depending on the sign of the rotation...

Reply:
Nice, thanks for the write up! But I meant to leave comments in the code haha :sweat_smile: This way future contributors can easily see the logic!

Reply:
Haha oops will do